include/pg_tmgr.h

```diff
@@ -0,0 +1,28 @@
+/*-------------------------------------------------------------------------
+ *
+ * include/pg_tmgr.h
+ *
+ * Transaction manager API
+ *
+ * Copyright (c) 2014-2015, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef PG_TMGR_H
+#define PG_TMGR_H
+
+typedef struct { 
+	bool (*Begin)(PGconn* conn);
+	bool (*Prepare)(PGconn* conn, char const* relationName, ShardId shardId);
```

Comment:
It might be worth calling "relationName" and "shardId" something like "prefix" and "transactionId". Since they are only used to give the transaction a name, they don't need to an actual relation name or shard id and we might want to use these functions for other purposes like multi-shard updates.